rllib/agents/ppo/ppo_torch_policy.py

```diff
@@ -68,9 +68,10 @@ def __init__(self,
             use_gae (bool): If true, use the Generalized Advantage Estimator.
         """
         if valid_mask is not None:
+            num_valid = torch.sum(valid_mask)
 
             def reduce_mean_valid(t):
-                return torch.mean(t * valid_mask)
+                return torch.sum(t * valid_mask) / num_valid
```

Comment:
This is the same operation right? Or was it fixing a bug?

Reply:
No this was a bug. If you have `False`s in your mask, the mean will not be the mean of only the valid values, but the False-(=0)-diluted mean.

Reply:
Oof

Reply:
It's ok. Torch LSTMs were never working anyways (hence this PR  😉) and the above code would only be executed in the RNN-case.

Reply:
Torch doesn't have tf.boolean_mask, so I "implemented" it at first the wrong way with `t * valid_mask` when I did the torch-PPO translation. But again, only affected the RNN case, which wasn't function until now.